refactor(devtools): improve route data and resolver views

SkyZeroZx opened the pull request:
Adds an enhanced route data tree view to better visualize both route resolvers and router data. Resolvers now include direct access to their source code for easier inspection and debugging

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
Currently, it does not display resolvers, and router data information, when relatively large, is not easy to read.
 
<img width="554" height="525" alt="image" src="https://github.com/user-attachments/assets/74bcc411-99b0-45d2-9e44-69413a4a2bb0" />



## What is the new behavior?
 
<img width="527" height="716" alt="image" src="https://github.com/user-attachments/assets/1f0252b9-4add-4deb-b349-181ba214b82a" />


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

hawkgs commented on devtools/projects/ng-devtools/src/lib/devtools-tabs/router-tree/route-data-tree/route-data-tree.component.scss:
```diff
@@ -0,0 +1,86 @@
+  }
+
+  .data-row {
+    $margin-left: 1.25rem;
```
I think we should reduce this to `0`. My idea is to align the indentation of the object to the other route details row values like the flags, for example.

hawkgs commented on devtools/projects/ng-devtools/src/lib/devtools-tabs/router-tree/route-details-row.component.html:
```diff
@@ -18,18 +18,21 @@
+      @if (rowValue() && renderValueAsJson()) {
+        <ng-route-data-tree
+          (viewSource)="actionBtnClick.emit($event)"
+          [showButton]="actionBtnType() !== 'none'"
```
Is there a reason why the check isn't `actionBtnType() === 'view-source'`?

SkyZeroZx replied:
If in the future we want to extend or use another type, I think that’s the first idea. The alternative would be to pass through the type and map it in the new component  , what do you think?

hawkgs replied:
That'll work as well, as long as the button is rendered when `type === 'view-source'`, because right now `ActionBtnType` is already composed of 3 types, so if we have something like this:

```html
<tr
  ng-route-details-row
  [renderValueAsJson]="true"
  actionBtnType="navigate"
  ...
/>
```

The "view source" button will be rendered, even though you provided `navigate` (which isn't supported in this case). 

Keep in mind that the whole action button stuff is a bit of a band-aid solution that we've introduced. It will probably be overhauled along with the `route-details-row`, so as long as the behavior is correct, I think we are good.

hawkgs commented on devtools/projects/ng-devtools/src/lib/devtools-tabs/router-tree/route-data-tree/route-data-tree.component.ts:
```diff
@@ -0,0 +1,94 @@
+      const obj: Record<string, any> = {};
+      data.forEach((item: any) => {
+        obj[item.key] = item.value;
+      });
```
Nit/style: We try to stick to `for...of` instead of `forEach`.

SkyZeroZx replied:
Updated

SkyZeroZx committed:
refactor(devtools): improve route data and resolver views

Adds an enhanced route data tree view to better visualize both route resolvers and router data.

hawkgs commented:
@SkyZeroZx, did you, by any chance, forget to push your changes since I don't see them in the diff?

hawkgs reviewed (approved):
LGTM ✅

Thanks!

thePunderWoman commented:
This PR was merged into the repository. The changes were merged into the following branches:

- main: 373c101d02de0cea3fccbbfdf61fcd7ea9c049b0
